broker/data-access-layer/iaas/src/AwsClient.js

```diff
@@ -0,0 +1,308 @@
+          .set('isTruncated', isTruncated)
+          .value()
+        ));
+        return files;
+      });
+  }
+
+  remove(container, file) {
+    if (arguments.length < 2) {
+      file = container;
+      container = this.containerName;
+    }
+    logger.info('Deleting file ' + file + ' from container ' + container);
+    return this.storage
+      .deleteObjectAsync({
+        Bucket: container,
+        Key: file
+      })
+      .catch(BaseCloudClient.providerErrorTypes.Unauthorized, err => {
+        logger.error(err.message);
+        throw new Unauthorized('Authorization at the cloud storage provider failed');
+      })
+      .catchThrow(BaseCloudClient.providerErrorTypes.NotFound, new NotFound(`Object '${file}' not found`));
+  }
+
+  download(options) {
+    return Promise
+      .try(() => this.storage.getObject(options).createReadStream())
+      .then(streamToPromise)
+      .catchThrow(BaseCloudClient.providerErrorTypes.NotFound, new NotFound(`Object '${options.Key}' not found`));
+  }
+
+  upload(options) {
+    let s3Settings = {
+      partSize: 5 * 1024 * 1024,
+      queueSize: 1
+    };
+    return this.storage.uploadAsync(options, s3Settings);
+  }
+
+  uploadJson(container, file, data) {
+    if (arguments.length < 3) {
+      data = file;
+      file = container;
+      container = this.containerName;
+    }
+    logger.info('Uploading file ' + file + ' to container ' + container);
+    return this
+      .upload({
+        Bucket: container,
+        Key: file,
+        Body: new Buffer(JSON.stringify(data, null, 2), 'utf8')
+      });
+  }
+
+  downloadJson(container, file) {
+    if (arguments.length < 2) {
+      file = container;
+      container = this.containerName;
+    }
+    logger.info('Downloading file ' + file + ' from container ' + container);
+    return this
+      .download({
+        Bucket: container,
+        Key: file
+      })
+      .then(data => {
+        const response = JSON.parse(data);
+        response.trigger = response.trigger === CONST.BACKUP.TRIGGER.MANUAL ? CONST.BACKUP.TRIGGER.SCHEDULED : response.trigger;
+        // The above conversion is done to handle existing CRON Jobs which set this trigger as 'manual' even for scheduled Jobs
+        // Above conversion can be removed and code changes can be revereted 14 days after the current fix goes live
+        return response;
+      })
+      .catchThrow(SyntaxError, new NotFound(`Object '${file}' not found`));
+  }
+
+  deleteSnapshot(snapshotId) {
+    return Promise
+      .try(() => {
+        return this.blockstorage
+          .deleteSnapshot({
+            SnapshotId: snapshotId
+          })
+          .promise()
+          .then(retval => {
+            logger.info(`Deleted snapshot ${snapshotId}`, retval);
+          })
+          .catch(err => {
+            logger.error(`Error occured while deleting snapshot ${snapshotId}`, err);
+            throw err;
+          });
+      });
+  }
+
+  static validateParams(options) {
```

Comment:
This function was added taking reference from `GcpClient`, to test whether necessary arguments are passed to create the client instance.